cli.js

```diff
@@ -181,29 +187,45 @@ if (require.main === module) {
                 type: 'string',
                 required: 'true',
                 default: hecate.port
-            }];
-
-            // if the username and password hasn't already been set, prompt for it
-            if (!hecate.user) {
-                args = args.concat(auth(hecate.user));
-            }
-
-            prompt.get(args, (err, res) => {
+            }], (err, res) => {
                 if (err) throw err;
-
                 hecate.url = res.url;
                 hecate.port = res.port;
-
-                if (res.hecate_username && res.hecate_password) {
-                    hecate.user = {
-                        username: res.hecate_username,
-                        password: res.hecate_password
-                    };
-                }
-
                 argv.cli = true;
 
-                return run();
+                // if a custom auth policy hasn't been passed
+                if (!hecate.auth_rules) {
+                    // fetch auth
+                    hecate.auth({}, (err, auth_rules) => {
+                        // if requesting auth returns a 401
+                        if (err.message === '401: Unauthorized') {
```

Comment:
oof we should get https://github.com/mapbox/HecateJS/pull/16 back into shape and merge it in. Not a blocker, just could see this causing issue as we return a slightly different error code depending on if it is the middleware (text response) vs an API level block (JSON response)

Reply:
Oh, I see the different response doesn't matter as I hadn't seen how you generated this code below. Worries abated, through we should probably pass through error code at some point anyway